### PR TITLE
Seprated code docs into `core` and `integration` docs

### PR DESCRIPTION
**Pull request by AlexejPenner**
## Describe changes
I tried to fix the github action for the API docs, let's hope it works in the next release.

While I was at it I also separated out the Integrations from the Core code.

Before:
before

New:
integration
core

## Pre-requisites
Please ensure you have done the following:
- [ ] I have read the **CONTRIBUTING.md** document.
- [ ] If my change requires a change to docs, I have updated the documentation accordingly.
- [ ] If I have added an integration, I have updated the [integrations](https://docs.zenml.io/component-gallery/integrations) table and the [corresponding website section](https://zenml.io/integrations).
- [ ] I have added tests to cover my changes.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Other (add details above)

### CODE REVIEW

**Comment by AlexejPenner** [comment]
@htahir1 just tagged you for your input on the overall idea of splitting it up into core/integration docs

@stefannica tagged you to have a look at the "fix" in the .github folder

@bcdurak tagged you as the lucky guy to review the code :)

File: docs/mkdocstrings_helper.py [hunk 1]
```diff
@@ -181,6 +193,21 @@ def generate_docs(
...
 
+    integration_file_contents = [INTEGRATION_DOCS_TITLE]
+    integrations_file_content = create_entity_docs(
+        api_doc_file_dir=integrations_dev_doc_file_dir,
+        ignored_modules=["__init__.py", "__pycache__"],
+        sources_path=path / 'integrations',
+        index_file_contents=integration_file_contents,
+        md_prefix='integrations'
+    )
+    # integrations_file_str = '\n'.join(integrations_file_content)
```
Comment (htahir1) [inline docs/mkdocstrings_helper.py:204]: Maybe take this out?

**Review by htahir1 — APPROVED** [review]
Left a small nit but otherwise LGTM!

File: docs/mkdocstrings_helper.py [hunk 2]
```diff
@@ -181,6 +193,21 @@ def generate_docs(
...
+    # to_md_file(
+    #     integrations_file_str,
+    #     'integrations.md',
+    #     out_path=output_path,
+    # )
```
Comment (AlexejPenner) [inline docs/mkdocstrings_helper.py:209]:
```suggestion
```

**Review by stefannica — APPROVED** [review]
The github action change LGTM, but I haven't a clue how to validate the mkdocsstrings_helper.py code changes

**Pull request merged**